### DISCUSSION TITLE
Testing system documentation

### DISCUSSION THREAD

**Discussion by rtaylo45** [idaholab/moose#19926] [Q&A Tools] [answered]
Hello,

Im trying to set up more advanced testing in my app and im finding it hard to find documentation on the moose testing system. For example [here](https://mooseframework.inl.gov/python/TestHarness.html), shows a list of testers but only two of the testers are linked to documentation. In those links they also don't really show any sort of arguments that can be added with these testers so that i can change the default behavior. Can someone point me to where i can find more documentation or more advanced usage i can use as examples?

Thanks,
Zack

**Comment by GiudGiud** [comment, marked as answer]
Hello

I'm not sure we have pages for each type of test. @loganharbour do you know if we have more docs on the civet side on this?
Some of the testers, like exodiff, are external and have their own docs. I could not find a page on mooseframeworks.org with docs on how to specify extra arguments either. 
The repository has a lot of examples, what specifically are you looking for?

Happy holidays,
Guillaume

**Reply by rtaylo45** [reply]
I was able to find an example in the repository for the specific test type i was looking for. I was trying to find a way to test for invalid input, but the question was meant to be more general. It was difficult for me to find out what i would even search for in the repository.

**Reply by rtaylo45** [reply]
@GiudGiud is there a way to flag a test to run only if a MakeFile variable is set?

**Reply by GiudGiud** [reply]
I don't think so. There's only the executable_pattern afaik for that kind of skip.
You could pass a command line argument to the test to make it react to a Makefile variable being set or not (with additional code in your application to react to this argument), but the result could still be affected, and whether to consider the test result would not

**Reply by GiudGiud** [reply]
I'm writing some docs for the testing system, I ll let you know if I find anything

**Reply by GiudGiud** [reply]
Please see this PR https://github.com/idaholab/moose/pull/19985 and let me know if there's more information you think I should include in the testers docs
I didnt see anything about a Makefile variable parameter for the testers. Maybe @milljm  or @loganharbour  will know about this
